### ISSUE TITLE
Mail::Address display_name loses UTF-8 encoding on formatting

### ISSUE THREAD

**Issue by SokratisVidros** [mikel/mail#810]
``` ruby
email = "eugene@example.com"
name = "Eugéne Gonçalves"

# Encoding is UTF-8
name.encoding # <Encoding:UTF-8>

ma = Mail::Address.new(email)
ma.display_name = name
ma.format

name # "Eug\xC3\xA9ne Gon\xC3\xA7alves"

# Encoding has changed to ASCII-8BIT
name.encoding # <Encoding:ASCII-8BIT>
```

The above code was tested with ruby 2.0.0-p353 and 2.0.0-p576. A workaround is to duplicate the name string before the assignment.

```
ma.display_name = name.dup
```

**Comment by Exoth** [comment]
Gettings the same problem. Then it breaks on deliver attempt, because such string can't be converted to utf-8 (getting Encoding::UndefinedConversionError: "\xC3" from ASCII-8BIT to UTF-8)

**Comment by Exoth** [comment]
Actually the problem is with the resulting formatted value as well.

**Comment by Exoth** [comment]
The problem should have been fixed by this commit:
https://github.com/mikel/mail/commit/9c7376a47962d6ac7cf0b84184971d09f022225f

**Comment by jeremy** [comment]
Fixed by #1107 

```ruby
email = "eugene@example.com"
name = "Eugéne Gonçalves"

# Encoding is UTF-8
name.encoding # <Encoding:UTF-8>

ma = Mail::Address.new(email)
ma.display_name = name
ma.format

name # "Eugéne Gonçalves"
name.encoding # <Encoding:UTF-8>
```
